Update SimpleHttpOperator to take auth type object

fredthomsen opened the pull request:
A `requests.auth.AuthBase` object is not passed through from the
`SimpleHttpOperator` to the underlying `HttpHook`, thus if you want to
use the `SimpleHttpOperator` but have a custom auth_type, you must
inherit from it and override the execute method.  Update the operator
to take this parameter.

Seems like a small oversight that existing tests should cover.

uranusjr commented on airflow/providers/http/operators/http.py:
```diff
@@ -84,6 +88,7 @@ def __init__(
         extra_options: Optional[Dict[str, Any]] = None,
         http_conn_id: str = 'http_default',
         log_response: bool = False,
+        auth_type: Any = HTTPBasicAuth,
```
```suggestion
        auth_type: Type[AuthBase] = HTTPBasicAuth,
```

See https://docs.python-requests.org/en/master/user/authentication/#new-forms-of-authentication

fredthomsen replied:
Woops done.

Commit:
Update `SimpleHttpOperator` to take auth object

A `requests.auth.AuthBase` object is not passed through from the
`SimpleHttpOperator` to the underlying `HttpHook`, thus if you want to
use the `SimpleHttpOperator` but have a custom auth_type, you must
inherit from it and override the execute method.  Update the operator
to take this parameter.

uranusjr commented on airflow/providers/http/operators/http.py:
```diff
@@ -84,6 +88,7 @@ def __init__(
         extra_options: Optional[Dict[str, Any]] = None,
         http_conn_id: str = 'http_default',
         log_response: bool = False,
+        auth_type: Type(AuthBase) = HTTPBasicAuth,
```
```suggestion
        auth_type: Type[AuthBase] = HTTPBasicAuth,
```

I’m surprised Mypy does not complain about this. Is this even legal?

fredthomsen replied:
Actually, mypy does not complain about this interestingly enough.  I was surprised.  However, I did not re-run the provider tests on this annotation update and they do indeed fail.
